napari/components/add_layers_mixin.py

```diff
@@ -155,91 +159,77 @@ def add_image(
+            'metadata': metadata,
+            'scale': scale,
+            'translate': translate,
+            'opacity': opacity,
+            'blending': blending,
+            'visible': visible,
+        }
+
         if channel_axis is None:
-            if colormap is None:
-                colormap = 'gray'
-            if blending is None:
-                blending = 'translucent'
-            layer = layers.Image(
-                data,
-                rgb=rgb,
-                is_pyramid=is_pyramid,
-                colormap=colormap,
-                contrast_limits=contrast_limits,
-                gamma=gamma,
-                interpolation=interpolation,
-                rendering=rendering,
-                iso_threshold=iso_threshold,
-                attenuation=attenuation,
-                name=name,
-                metadata=metadata,
-                scale=scale,
-                translate=translate,
-                opacity=opacity,
-                blending=blending,
-                visible=visible,
-            )
-            self.add_layer(layer)
-            return layer
+            kwargs['colormap'] = kwargs.get('colormap', 'gray')
+            kwargs['blending'] = kwargs.get('blending', 'translucent')
+            return self.add_layer(layers.Image(data, **kwargs))
         else:
-            if is_pyramid:
-                n_channels = data[0].shape[channel_axis]
-            else:
-                n_channels = data.shape[channel_axis]
-
-            name = ensure_iterable(name)
-
-            if blending is None:
-                blending = 'additive'
-
-            if colormap is None:
-                if n_channels < 3:
-                    colormap = colormaps.MAGENTA_GREEN
+            n_channels = (data[0] if is_pyramid else data).shape[channel_axis]
+            kwargs['blending'] = kwargs.get('blending', 'additive')
+
+            # turn the kwargs dict into a mapping of {key: iterator}
+            # so that we can use {k: next(v) for k, v in kwargs.items()} below
+            for key, val in kwargs.items():
+                if key in ('scale', 'translate', 'contrast_limits'):
+                    kwargs[key] = iter(
+                        ensure_iterable_sequences(val, n_channels)
+                    )
+                elif key == 'metadata':
+                    if val is None or isinstance(val, dict):
+                        kwargs[key] = itertools.repeat(metadata)
+                    elif (
+                        isinstance(val, list)
+                        and len(val) == n_channels
+                        and all(isinstance(m, dict) for m in val)
+                    ):
+                        kwargs[key] = iter(val)
+                    else:
+                        raise ValueError(
+                            "metadata must either be a dict, or a list of dicts "
+                            "with a length equal to the number of channels."
+                        )
```

Comment:
Why is this not raised with the other maybe-sequences?

Reply:
changed `ensure_sequence_of_iterables` to handle dicts too.